simulation/amaru-sim/src/simulator/run.rs

```diff
@@ -315,3 +284,72 @@ impl BlockFetcher for FakeBlockFetcher {
         Ok(vec![])
     }
 }
+
+/// This implementation of ForwardEventListener sends the received events a Sender that can collect them
+/// to check them later.
+///
+/// A message id is maintained for each received event and each message is duplicated to the number of downstream peers.
+#[derive(Clone, Debug)]
+pub struct MockForwardEventListener {
+    node_id: String,
+    number_of_downstream_peers: u8,
+    sender: mpsc::Sender<Envelope<ChainSyncMessage>>,
+    msg_id: Arc<Mutex<u64>>,
```

Comment:
should use `Arc<AtomicU64>` for such functionality (with `Ordering::Relaxed` because only the counter needs to be consistent)